app/models/subnet.rb

```diff
@@ -279,12 +279,11 @@ def subnet_for(ip)
     end
 
     # This casts Subnet to Subnet::Ipv4 if no type is set
-    def new_with_default_type(*attributes, &block)
+    def new(*attributes, &block)
       type = attributes.first.with_indifferent_access.delete(:type) if attributes.first.is_a?(Hash)
-      return Subnet::Ipv4.new_without_cast(*attributes, &block) if self == Subnet && type.nil?
```

Comment:
Wasn't there the `new_without_cast` for a reason there? Why wasn't there just `Subnet::Ipv4.new(*attributes, &block)` before?

Reply:
I think that's also because of the change in app/models/concerns/foreman/sti.rb

Reply:
As long as the tests are green, I'm ok with that, just wondering if there isn't something hidden around this.